src/test/java/BaseTest.java

```diff
@@ -1,11 +1,166 @@
+    public static Actions actions = null;
+    public static String url = "https://qa.koel.app/";
+
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+    @BeforeMethod
+    @Parameters ({"BaseURL"})
+    public void launchBrowser(String BaseURL) throws InterruptedException{
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        url = BaseURL;
+        driver.get(url);
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        actions = new Actions(driver);
+
+    }
+    @AfterMethod
+    public void logout() {
+
+        driver.quit();
+    }
+    @DataProvider (name="CorrectLoginProviders")
+    public static Object[][] getDataFromDataProviders(){
+        return new Object[][] {
+                {"matt.pierce@testpro.io", "te$t$tudent"}
+        };
+    }
+
+    public static void openLoginUrl() {
+        String url = "https://qa.koel.app/";
+        driver.get(url);
+    }
+
+    protected static void enterEmail(String email) throws InterruptedException{
+        WebElement emailInput = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='email']")));
+        emailInput.click();
+        emailInput.clear();
+        //inputs valid email address into email input field
+        emailInput.sendKeys(email);
+    }
+
+    protected static void enterPassword(String password) throws InterruptedException{
+
+        WebElement passwordInput = wait.until(ExpectedConditions.elementToBeClickable(By.cssSelector("[type='password']")));
+        passwordInput.clear();
+        //inputs valid password for login into password field
+        passwordInput.sendKeys(password);
+    }
+
+    public void clickSubmit() throws InterruptedException{
+        WebElement loginButton = wait.until(ExpectedConditions.elementToBeClickable((By.cssSelector("[type='submit']"))));
+        loginButton.click();
+    }
+
+    @Test
+    public void addSongToPlaylist(String email, String password) throws InterruptedException{
+        String newSongAddedNotificationText = "Added 1 song into";
+        //open url
+        //openLoginUrl();
+        //input email
+        enterEmail(email);
+        //input password
+        enterPassword(password);
+        //click submit
+        clickSubmit();
+
+        //search for song
+        searchForSong("Dark Days");
+        //click View All Btn
+        clickViewAllBtn();
+        // verify on search results page
+        verifySearchUrl();
+        //select the first song returned in the search
+        selectFirstSongResult();
+        //click Add To Btn
+        clickAddToBtn();
+        //choose Playlist to add the song to
+        choosePlaylist();
+        //checking for success message notification
+        Assert.assertTrue((getNotificationText().contains(newSongAddedNotificationText)));
+
+
+    }
```

Comment:
Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/e08645bb-c5e8-42bc-a51c-5fa9e4dd6666)

HINT

**HINT**

1. **BasePage and BaseTest** 

**BaseTest** — Implements the necessary methods for work with webdriver.

The BaseTest class contains common setup and teardown methods, as well as any shared functionality or configurations required by multiple test classes.

**BaseTest**

The problem we are trying to solve here is the reuse of common behaviors across test classes avoiding coding duplication and centralizing these actions at one point.

**@BeforeSuite**	- The annotated method will be run before all tests in this suite have run.

**@BeforeMethod** -	The annotated method will be run before all the test methods in the current class have been run.

**@AfterMethod**	- The annotated method will be run after each test method.

_Example without a Base Test Class_

![image](https://user-images.githubusercontent.com/104761482/233972627-2d7ffdc7-6a2f-4e9f-8a3a-a1b4a7a47bc4.png)


**BasePage**  — class serves as a foundation or parent class for individual page classes. It contains common methods and functionalities that are shared across multiple pages in an application.

_Example without a Base Page Class_

![image](https://user-images.githubusercontent.com/104761482/236891493-7f7013c7-fde2-4bf9-9084-3aa7483df8e3.png)


2. **Test in Page Object Model (POM)**

![Clean_Test](https://user-images.githubusercontent.com/104761482/236863813-f30d7bfc-e89d-4419-bdea-4b1517155e39.png)